monai/apps/auto3dseg/bundle_gen.py

```diff
@@ -44,74 +42,32 @@ class BundleAlgo(Algo):
-        │   ├── network.yaml  # automatically generated network yaml from a set of ``template_configs``
-        │   ├── transforms_train.yaml  # automatically generated yaml to define tranforms for training
-        │   ├── transforms_validate.yaml  # automatically generated yaml to define transforms for validation
-        │   └── transforms_infer.yaml  # automatically generated yaml to define transforms for inference
-        └── scripts
-            ├── algo.py
-            ├── infer.py
-            ├── train.py
-            └── validate.py
+    ``export_to_disk``. Then created bundle keeps the same file structure as the template.
 
     """
 
-    def __init__(self, template_configs=None, scripts_path=None, meta_data_filename=None, parser_args=None):
+    def __init__(self, template_path: str):
         """
-        Create an Algo instance based on a set of bundle configuration templates and scripts.
+        Create an Algo instance based on the predefined Algo template.
 
         Args:
-            template_configs: a json/yaml config file, or a folder of json/yaml files.
-            scripts_path: a folder to python script files.
-            meta_data_filename: optional metadata of a MONAI bundle.
-            parser_args: additional input arguments for the build-in ConfigParser ``self.cfg.read_config``.
-        """
-        if os.path.isdir(template_configs):
-            self.template_configs = []
-            for ext in ("json", "yaml"):
-                self.template_configs += glob(os.path.join(template_configs, f"*.{ext}"))
-        else:
-            self.template_configs = template_configs
-        self.meta_data_filename = meta_data_filename
-        self.cfg = ConfigParser(globals=False)  # TODO: define root folder (variable)?
-        if self.template_configs is not None:
-            self.load_templates(self.template_configs, meta_data_filename, parser_args)
-
-        self.scripts_path = scripts_path
-        self.data_stats_files = None
-        self.data_list_file = None
-        self.output_path = None
-        self.name = None
-        self.best_metric = None
+            template_path: path to the root of the algo template.
 
-    def load_templates(self, config_files, metadata_file=None, parser_args=None):
         """
-        Read a list of template configuration files
-
-        Args:
-            config_file: bundle config files.
-            metadata_file: metadata overriding file
-            parser_args: argument to parse
 
-        """
-        parser_args = parser_args or {}
-        self.cfg.read_config(config_files, **parser_args)
-        if metadata_file is not None:
-            self.cfg.read_meta(metadata_file)
+        self.template_path = template_path
+        self.data_stats_files = ""
+        self.data_list_file = ""
+        self.output_path = ""
+        self.name = ""
+        self.best_metric = None
```

Comment:
Can we have an additional list variable here? I would suggest to use such variable to keep track of all the autogenerated parameters in the config files. Example implementation can be found [here ](https://github.com/Project-MONAI/research-contributions/blob/974b890b19dec456c5115ad2881ef368ecfcd969/auto3dseg/algorithm_templates/segresnet2d/scripts/algo.py#L112)
```python
self.fill_records = []
```

Reply:
Sure, added the logic in the latest commit.

Thanks.